### ISSUE TITLE
Enable multiple treesitter grammars in `shell.nix`

### ISSUE THREAD

**Issue by armijnhemel** [ngi-nix/ngipkgs#139]
I have trouble figuring out how to enable multiple grammars for treesitter, which I want to use (preferably with a `shell.nix` as that is what I am comfortable with).

I would appreciate some advise for both a "clean" repository as well as for this one:

https://github.com/armijnhemel/proximity_matcher_webservice/

Background information: https://nlnet.nl/project/AutomatedClearing/

**Comment by fricklerhandwerk** [comment]
@armijnhemel Please repost your question on the [NixOS Discourse](https://discourse.nixos.org/c/learn/9), as that may get you the answer more quickly.

**Comment by armijnhemel** [comment]
> @armijnhemel Please repost your question on the [NixOS Discourse](https://discourse.nixos.org/c/learn/9), as that may get you the answer more quickly.

Thank you, I will do that!

**Comment by armijnhemel** [comment]
https://discourse.nixos.org/t/need-help-enabling-grammars-in-treesitter-python/39500

No replies yet though ;-)

**Comment by mightyiam** [comment]
Hey, @armijnhemel. You still need help with this? I'll also try to PM you.

**Comment by armijnhemel** [comment]
> Hey, @armijnhemel. You still need help with this? I'll also try to PM you.

Yes. I sent you a reply a few days ago, which I will quote here:

"The ideal outcome for me wouldn't just be a fix for my particular problem, but also some docs that can for example be shared on the NixOS wiki, or some expressions in nixpkgs."

**Comment by stepbrobd** [comment]
We've implemented a hacky solution that should unblock you, but it only works with tree-sitter (python binding) < 0.22.0

Here's the link: <https://github.com/ngi-nix/nixpkgs/blob/c06e35046b115819be5e6350e5485135605df942/pkgs/development/python-modules/tree-sitter-languages/default.nix>

**Comment by mightyiam** [comment]
https://github.com/NixOS/nixpkgs/pull/320783

**Comment by fricklerhandwerk** [comment]
@armijnhemel this turned out to be a major infrastructure contribution wich is now done, please check the documentation in the PR: https://github.com/NixOS/nixpkgs/pull/320783/files
